### DISCUSSION TITLE
Code Chunk Runs, but Render Fails. Help?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#165] [Q&A] [answered]
There's a pivot_longer in a code chunk I can run just fine, but when I knit, error keeps poppin out. Help?
![Capture](https://user-images.githubusercontent.com/95233522/176739274-2faf3ee1-d535-4f5c-85f5-926ae35bc457.JPG)

**Comment by onlyphantom** [comment, marked as answer]
`predict` is a function in R (from `stats`):

```
function (object, ...) 
UseMethod("predict")
<bytecode: 0x562cd3119ce0>
<environment: namespace:stats>
```

When you knit, `pivot_longer` rightly gives you the error saying `no applicable method for 'pivot_longer' applied to an object of class "function"`. If you try and do `pivot_longer(head)` you get the same thing, since `head` is also a function and pivot longer doesn't know what to do with it. 

But why is it that when you run in your code chunk it works? Because you overwrite `predict` to point to another variable by doing assignment:
```
# don't do this
predict <- c(1,2,5)
# don't do this
head <- c(1,2,5)
```
So when you run your chunk, in that specific environment where you've overwritten `predict` and make it point to a vector, it no longer is a function. But when you knit it from top to bottom, that overwrite isn't there, or isn't placed before line 787, so it didn't get execute before it. 

So when you knitting hits line 787, `predict` is still a function, just like `head` is, and therefore throws an error. Generally it's considered bad form to use these variable names, it leads to namespace pollution and also make your code unpredictable. 

Hope this helps!

**Reply by Hertzspr** [reply]
Thank you, that helps.
